components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java

```diff
@@ -445,10 +445,14 @@ public static OMElement buildOMElement(InputStream inputStream) throws Exception
         try {
             connection = APIMgtDBUtil.getConnection();
             statement = connection.createStatement();
-            String query;
+            String query, mssqlQuery;
 
             query = "SELECT  CONSUMER_KEY,NAME FROM AM_APPLICATION_KEY_MAPPING  NATURAL JOIN AM_APPLICATION  NATURAL JOIN AM_SUBSCRIBER WHERE USER_ID = '" + subscriberName + "' ";
+            mssqlQuery = "SELECT CONSUMER_KEY, NAME FROM AM_APPLICATION_KEY_MAPPING INNER JOIN AM_APPLICATION ON AM_APPLICATION_KEY_MAPPING.APPLICATION_ID=AM_APPLICATION.APPLICATION_ID INNER JOIN AM_SUBSCRIBER ON AM_APPLICATION.SUBSCRIBER_ID = AM_SUBSCRIBER.SUBSCRIBER_ID WHERE AM_SUBSCRIBER.USER_ID = '" + subscriberName + "' ";
 
+            if(connection.getMetaData().getDatabaseProductName().contains("Microsoft")){
```

Comment:
Also think about properly abstracting out DAO layers to support multiple database types as adding if-else logics to separate out database logics can often make your code not too readable as well as complex.